crates/uv/src/commands/project/init.rs

```diff
@@ -392,6 +424,7 @@ pub(crate) enum InitProjectKind {
     #[default]
     Application,
     Library,
+    Script,
```

Comment:
I think we should use two nested enums here, to make the `anyhow::bail!("Error during script initialization")` below impossible. So, something like:

```rust
#[derive(Debug, Copy, Clone, Default)]
pub(crate) enum InitKind {
    #[default]
    Project(InitProjectKind),
    Script,
}

#[derive(Debug, Copy, Clone, Default)]
pub(crate) enum InitProjectKind {
    #[default]
    Application,
    Library,
}
```

Reply:
Agreed, this change made many things much cleaner :)